### ISSUE TITLE
Support for azurerm_postgresql_flexible_server version 16

### ISSUE THREAD

**Issue by davidkarlsen** [hashicorp/terraform-provider-azurerm#24049]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Description

The provider shold allow for version 16 which is a valid version.

### New or Affected Resource(s)/Data Source(s)

azurerm_postgresql_flexible_server

### Potential Terraform Configuration

```hcl
version to include 16 as allowed value,

https://learn.microsoft.com/en-us/azure/postgresql/flexible-server/concepts-supported-versions
```

**Comment by neil-yechenwei** [comment]
Thanks for raising this issue. It has been added to TF and will be released to v3.83.0. See more details from https://github.com/hashicorp/terraform-provider-azurerm/pull/24035.

**Comment by tombuildsstuff** [comment]
Fixed via #24016
